ScrollToTop: Add OnClick event

i-n-n-k-e-e-p-e-r opened the pull request:
## Description
Closes #6517
MudScrollToTop it is an invisible wrapper for all inner components is does not allow 'clicking' them. So i think that better way to handle clicking is to add event handler for MudScrollToTop;

### Before

https://github.com/MudBlazor/MudBlazor/assets/25841638/d702fc1d-09a1-49c9-9276-516abfab5067

### After

https://github.com/MudBlazor/MudBlazor/assets/25841638/6b5e5c00-5809-4aec-b2e6-7523f3c9bf3d


## How Has This Been Tested?
Added new unit and visual tests for MudScrollToTop component.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)




## Checklist:
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] The PR is submitted to the correct branch (`dev`).
- [x] My code follows the code style of this project.
- [x] I've added relevant tests.

i-n-n-k-e-e-p-e-r committed:
Added OnClick event for MudScrollToTop (#6517)

+ MudScrollToTop it is an invisible wrapper for all inner components is does not allow 'clicking' them. So i think that better way to handle clicking is to add event handler for MudScrollToTop.
+ Added tests for MudScrollToTop;

Mr-Technician commented on src/MudBlazor/Components/ScrollToTop/MudScrollToTop.razor.cs:
```diff
@@ -118,9 +120,10 @@ private async void ScrollListener_OnScroll(object sender, ScrollEventArgs e)
         /// Scrolls to top when clicked
         /// </summary>
-        private void OnClick()
+        private async void OnElementClick(MouseEventArgs args)
```
Maybe call this OnButtonClick?

It should also be `async Task` rather than `async void`.

i-n-n-k-e-e-p-e-r replied:
It may be named OnButtonClick. I just thought that it is not a button, but just a span element. 
Oh. It's definitely should be Task. Thank you!

i-n-n-k-e-e-p-e-r replied:
Done

Mr-Technician replied:
> It may be named OnButtonClick. I just thought that it is not a button, but just a span element. Oh. It's definitely should be Task. Thank you!

This is a valid point, but the span wrapper almost always contains a Button as part of the scroll to top, so I don't think the name is inappropriate here.

Mr-Technician commented on src/MudBlazor.UnitTests.Viewer/TestComponents/Scroll/ScrollToTopTest.razor:
```diff
@@ -0,0 +1,46 @@
+
+<MudStack Row="true">
+    <button type="button" @onclick="ScrollBottom">Scroll bottom</button>
+    <MudText id="scroll_on_top_scroll_event_rised" Typo="Typo.subtitle1" Align="Align.Center">@scrolled</MudText>
```
"rised" is a typo, it should be "raised."

i-n-n-k-e-e-p-e-r replied:
Done

i-n-n-k-e-e-p-e-r committed:
Fixed return type of OnButtonClick + typo fixed.

+ Renamed MudScrollToTop.OnElementClick to MudScrollToTop.OnButtonClick;
+ Return type of MudScrollToTop.OnButtonClick (void -> Task) changed;
+ Typo fix;

henon commented on src/MudBlazor.UnitTests.Viewer/TestComponents/Scroll/ScrollToTopTest.razor:
```diff
@@ -0,0 +1,46 @@
+
+    async Task ScrollBottom()
+    {
+        await JS.InvokeVoidAsync("mudScrollManager.scrollIntoView", "h3[id='scroll_to_top_bottom_text']", "smooth");
```
We have a rule to not directly invoke JS except in tests testing the JS API. This makes sure the JS API usage is centralized in one point. Please add the appropriate function in ScrollManager or if it is already there then use that.

ScarletKuro replied:
I think it should be there?
https://github.com/MudBlazor/MudBlazor/blob/68c2d2ee0ed1256756e0c175ca4d13bbbf872f68/src/MudBlazor/Services/Scroll/ScrollManager.cs#L23

i-n-n-k-e-e-p-e-r replied:
Thanks a lot for the explanation! Corrected.

i-n-n-k-e-e-p-e-r committed:
Direct JS call replaced with ScrollManager method call

Direct JS API call replaced with ScrollManager method call in visual test for ScrollToTop component

henon reviewed (approved):
All yours @Mr-Technician

Mr-Technician commented on src/MudBlazor.UnitTests.Viewer/TestComponents/Scroll/ScrollToTopTest.razor:
```diff
@@ -0,0 +1,46 @@
+    public static string __description__ = "Scroll down. MudScrollToTop element must appear. After clicking fab button view must scroll to the top and first line of text must change.";
+   
+    string text = "Scroll and click fab, this text must change afterwards.";
+    string clicked = "false";
```
I'm being picky but I would rather see a Boolean used here instead of a string in HTML - that is more in line with what is done elsewhere.

Mr-Technician replied:
The same goes for scrolled. You can make both of them public and directly call `Should().BeTrue()` or `BeFalse()` from the test.

i-n-n-k-e-e-p-e-r replied:
It's not picky it's pretty logical. Thanks.

i-n-n-k-e-e-p-e-r committed:
Types of test parameters changed string->bool

Mr-Technician commented:
Thank you! @i-n-n-k-e-e-p-e-r
